### PR TITLE
Risk mode gw

### PR DESCRIPTION
**Pull request by GundulaW**

### CODE REVIEW

File: flood_adapt/object_model/hazard/hazard.py [hunk 1]
```diff
@@ -128,78 +177,260 @@ def add_discharge(self):
...
+        # Preprocess all hazard model input
+        self.preprocess_sfincs()
+        # add other models here
+
+    def postprocess_models(self):
+        # Preprocess all hazard model input
+        self.postprocess_sfincs()
+        # add other models here
 
     def run_models(self):
```
Comment (panosatha) [inline flood_adapt/object_model/hazard/hazard.py:198]: shouldn't there be a seperate run_sfincs method that is run here? The same way you do for preprocess for example

File: flood_adapt/object_model/scenario.py [hunk 2]
```diff
@@ -57,13 +57,34 @@ def save(self, filepath: Union[str, os.PathLike]):
...
+            self.direct_impacts.hazard.run_models()
         if not self.direct_impacts.has_run:
             self.direct_impacts.run_models()
         else:
             print(
                 f"Direct impacts for scenario '{self.attrs.name}' has already been run."
             )
+
+        if not self.direct_impacts.hazard.has_run:
+            self.direct_impacts.hazard.postprocess_models()
```
Comment (panosatha) [inline flood_adapt/object_model/scenario.py:83]: isn't this needed before you run direct impacts? At least for the risk run where the RP flood maps are needed as input

**Comment by panosatha** [comment]
check ruff, typos and TeamCity tests

**Review by panosatha — CHANGES_REQUESTED** [review]
Good job! I added some comments that you can have a look before merging

File: flood_adapt/object_model/scenario.py (hunk 2 above)
Comment (GundulaW) [reply flood_adapt/object_model/scenario.py:83]: I included all functions in one if statement: 

        if not self.direct_impacts.hazard.has_run:
            self.direct_impacts.hazard.preprocess_models()
            self.direct_impacts.hazard.run_models()
            self.direct_impacts.hazard.postprocess_models()
        else:
            print(f"Hazard for scenario '{self.attrs.name}' has already been run.")

**Pull request merged**